### PR TITLE
Remove word pair from address generator seed string

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem
Stake account address generation relies on a word-pair string that is unnecessary and may not be known to the owner of the system account.

#### Summary of Changes
Remove the word-pair string from address generation so the only information needed to create/retrieve a derived address is the system account pubkey and a numeric index.

Fixes #

### CODE REVIEW

**Comment by danpaul000** [comment]
This changes leaves the `name` in the StakerInfo struct without any use in the code (https://github.com/solana-labs/solana/blob/master/genesis/src/genesis_accounts.rs#L37)  .  Is there any reason to leave these in place for an internal coordination/anonymization or should I remove this field from all the StakerInfos?  The ValidatorInfos contain a useful `name` field for their vote account creation can be associated with a real validator entity.

**Comment by danpaul000** [comment]
> Safe to remove the `name` field from `StakerInfo` too?

@garious yes should be safe to do so.  Looks like we had the same comment at the same time.  I just wasn't sure if we had other reasons to retain the `name`s.  Seems not, so I'll remove them.

**Comment by garious** [comment]
Double-check with @rob-solana, but I don't see the need for the word pair. The public key is sufficient for joining rows in the spreadsheets.

**Comment by rob-solana** [comment]
I think we should leave the word pair unless you want to require a different staker key for every line in the stake holders' spreadsheet, or unless the spreadsheet is going to be reworked to have a single line per owner. The name value pair can be shared with the owners...

**Comment by rob-solana** [comment]
an issue arises when a single entity (staker, system account key) has multiple lines in the spreadsheet and the lines have different lockup terms

**Comment by rob-solana** [comment]
the noun-verb pair also lets us more easily proof-read genesis against the spreadsheet

**Comment by garious** [comment]
@rob-solana, a different key per spreadsheet line makes perfect sense to me

**Comment by danpaul000** [comment]
@rob-solana / @garious what is the action here?  I don't have visibility to the spreadsheet so can't help on that front.  It doesn't cost us anything to leave the word pairs in place for now until the spreadsheet can be reconciled, but only the removal of those word-pairs from the address seeding is in the blocking path today.

**Comment by garious** [comment]
@danpaul000 Yeah, leave the word-pairs, just don't use them in genesis.  To do that, consider leaving everything as it was, but just pass an empty string to the address generator.

**Comment by danpaul000** [comment]
@garious should be good now.  We can't pass a truly empty string to the address generator, we still need the iterator index, which plays nicely with the account derivation method we discussed the other day.

File: genesis/src/address_generator.rs [hunk 1]
```diff
@@ -3,28 +3,21 @@ use solana_sdk::{pubkey::Pubkey, system_instruction::create_address_with_seed};
...
 pub struct AddressGenerator {
     base_pubkey: Pubkey,
-    base_seed: String,
     program_id: Pubkey,
     nth: usize,
 }
 
 impl AddressGenerator {
-    pub fn new(base_pubkey: &Pubkey, base_seed: &str, program_id: &Pubkey) -> Self {
+    pub fn new(base_pubkey: &Pubkey, program_id: &Pubkey) -> Self {
```
Comment (mvines) [inline genesis/src/address_generator.rs]: Retaining support for base_seed seems nice, how about:
```suggestion
    pub fn new(base_pubkey: &Pubkey, base_seed: Option<&str>, program_id: &Pubkey) -> Self {
```

Comment (mvines) [reply genesis/src/address_generator.rs]: Oh, nm.  I see we won't share this code with `solana create-address-with-seed` so whatever

**Pull request merged**